mojap_airflow_tools/operators.py

```diff
@@ -150,6 +151,7 @@ def basic_kubernetes_pod_operator(
             security_context={
                 "allowPrivilegeEscalation": False,
                 "runAsNonRoot": True,
+                "runAsUser": randint(1000, 9999),
                 "privileged": False,
```

Comment:
(Also to fess up that's where I'm sourcing this solution from also! - I found the error by using `kubectl` to inspect the pods.)

Reply:
What about if someone has explicitly set a user in their Dockerfile with `USER <uid>` with custom permissions (ideally restricted, or otherwise)? This will overwrite that and could cause unexpected behaviour within the container.

I think we should allow `run_as_user` to be set via an optional argument, with default `None`. If `None`, `runAsUser` should not be included in the `security_context`. Where possible, we should encourage users to set a non-root user in their Dockerfile as they will have much more control over the permissions. This is something we could build into any Dockerfile template we provide to users.

Reply:
Perhaps there's a way to check that an image will run as a non-root user in the build, tag, push workflow? I.e., we enforce it for all user-created images?